core/unix/signal_linux.c

```diff
@@ -364,6 +364,8 @@ handle_pre_extended_syscall_sigmasks(dcontext_t *dcontext, kernel_sigset_t *sigm
     if (sizemask != sizeof(kernel_sigset_t))
         return false;
     ASSERT(!info->pre_syscall_app_sigprocmask_valid);
+    if (sigmask == NULL)
+        return true;
```

Comment:
I can initialize with false, sure. Though I thought it's ok since there is a clean contract between the caller and this function, and this function is not used anywhere else but these specific syscalls. I'll wait for whether you want to re-factor again as pointed out below.

Reply:
> there is a clean contract between the caller and this function

There's no documentation (i.e., comment in the declaration or definition) so I don't see where this contract is defined?  If new code is added how does it know this special behavior?